source/common/dynamo/dynamo_filter.cc

```diff
@@ -208,4 +211,26 @@ void DynamoFilter::chargeFailureSpecificStats(const Buffer::Instance& data) {
   }
 }
 
+void DynamoFilter::chargeTablePartitionIdStats(const Buffer::Instance& data) {
+  // Only log partition stats for single table operations.
+  if (table_descriptor_.table_name.empty() || operation_.empty()) {
+    return;
+  }
+  std::string body = buildBody(encoder_callbacks_->encodingBuffer(), data);
```

Comment:
you are potentially going to buildBody() multiple times I think, and now this is always going to happen in general. Let's make sure this only happens once.